src/FastSerialization/SegmentedMemoryStreamWriter.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace FastSerialization
+{
+    public class SegmentedMemoryStreamWriter : IMemoryStreamWriter
```

Comment:
❔ Can we avoid adding this for now, in favor of using the one from #307 based on memory-mapped files? I would send a separate pull request for the updated streams.